zebedee/collection.go

```diff
@@ -113,3 +115,38 @@ func (z *zebedeeClient) UpdateCollection(s Session, desc CollectionDescription)
 
 	return z.executeRequestNoResponse(req, http.StatusOK)
 }
+
+//UpdateCollectionContent updates content within a collection
+//  overwriteExisting (default:true) - if set to false, any existing content will not be overwritten
+//  recursive (default:false) - if set to true, all associated files alongside the page will be added to the collection's in progress directory
+//              if set to false, only the data.json file will be added to the collection's in progress directory
+//  validateJson (default:true) - if set to true, the json will be validated to ensure it's a valid page JSON structure
+func (z *zebedeeClient) UpdateCollectionContent(
+	s Session,
+	id, contentUri string,
+	content io.Reader,
+	overwriteExisting, recursive, validateJson bool) error {
+
+	url := fmt.Sprintf("%s/content/%s?uri=%s&overwriteExisting=%t&recursive=%t&validateJson=%t",
+		z.Host, id, contentUri, overwriteExisting, recursive, validateJson)
+
+	req, err := http.NewRequest(http.MethodPost, url, content)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("content-type", "application/json")
+	req.Header.Set(request.FlorenceHeaderKey, s.ID)
```

Comment:
I had initially tried to use this, but then didn't because of the io.reader usage - see point below. I have used `z.newAuthenticatedRequest()` for all other functions that follow this PR

Reply:
This function is now using `newAuthenticatedRequest`